tools/decompile.py

```diff
@@ -52,7 +52,7 @@ def __init__(self, nonmatching_path):
         self.text_offset = split[split.index("nonmatchings") + 1]
 
         assumed_path = "/".join(split[split.index("nonmatchings") + 1 : -1]) + ".c"
-        c_paths = [src for src in src_files if src.endswith(assumed_path)]
+        c_paths = [src for src in src_files if src.endswith(assumed_path) and self.overlay_name in src.split("/")]
```

Comment:
While this seems on the surface as though it would fix the issue, I'm not certain that it will behave in this manner in all circumstances.  My primary concern is that this could change an explicit error into a silently skipped condition and lead to ambiguous behavior.

I can't point to a specific instance where this would be the case, so I will leave it up to the judgement of the maintainers.